lib/pharos/cluster_manager.rb

```diff
@@ -40,30 +40,47 @@ def validate
       addon_manager.validate
     end
 
+    # @return [Pharos::Configuration::Host]
+    def preferred_master_host
+      master = config.master_hosts.find { |h|
+        ssh = ssh_manager.client_for(h)
+        ssh.file('/etc/kubernetes/kubelet.conf').exist?
+      }
+      if master
+        master
+      else
+        config.master_host
+      end
+    end
+
     def apply_phases
+      master = preferred_master_host
```

Comment:
@SpComb wdyt? This allows to replace the "first" master host.

Reply:
I mean: first master is not always the best one...

Reply:
Sure, although if you're using `api.endpoint` then it doesn't really matter which host you select... this is more explicit in #278

:-1: for using SSH to check for the `/etc/kubernetes/kubelet.conf` though... why not just use `https://#{host.address}:6443/healthz` to select a working master instead?

Reply:
We could check api health.. but it's not guaranteed to have any healthy master. Still we should prefer a master that has generated config (because of etcd, ca, etc..)

Reply:
> Still we should prefer a master that has generated config (because of etcd, ca, etc..)

But the `master` is not used for that, that uses whatever host the `ConfigureSecretsEncryption`, `ConfigureEtcdCa` etc phases run on first.

Reply:
> We could check api health.. but it's not guaranteed to have any healthy master

Having the kube-apiserver `/healthz` return HTTP 200 ok is more likely to indicate a healthy master than checking for the existence of `/etc/kubernetes/kubelet.conf`?